### PR TITLE
Adding Matisse example and making additions accordingly

### LINKED ISSUE #24: Plot command
Add something to the plot command for big graphs. Maybe a "plot only the edges" version for Sarah's leaf example

### PR DESCRIPTION
**Pull request by lizliz**
## Description

In the process of adding the Matisse example from Dan, I updated a bunch of stuff.

## Motivation and Context

* Added an iterator so that we can add a new vertex without giving it a name, and instead just grabbing the next available number/letter. This command also has a version to get the next $k$ letters since that was useful for the cycle bit. 
* Added a command to add a list of coordinates in a cycle. Again, since we were doing this for Dan's leaves, it seems like a useful command to have anyway. 
* Added min-max centering. In this case rather than having the mean of the vertex coordinates, I made it so that the min/max x-coords and the min/max y-coords are centered. 
* Added PCA transformation, again since Dan has been using this command a lot now it's intrinsic. 
* Updated the drawing so that you can scale and un-label vertices. This should close #24.
* Added option to the centering to either use mean or min_max, depending on the context.
* Of course, added Dan's Matisse example to the documentation so it will show up in the webpage as well. 

## How has this been tested?

Added tests that seemed reasonable. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [X] My code follows the code style of this project. (`make clean`)
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly. (`make html`)
- [X] I have incremented the version number in the `pyproject.toml` file.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed. (`make tests`)

### CODE REVIEW

File: ect/ect/embed_graph.py [hunk 1]
```diff
@@ -53,6 +61,50 @@ def add_nodes_from(self, nodes, coordinates):
         super().add_nodes_from(nodes)
         self.coordinates.update(coordinates)
 
+    def next_vert_name(self, s, num_verts=1):
```
Comment (Yemeen) [inline ect/ect/embed_graph.py:64]: can we separate into type checking then increment integers or running a sub function `increment_string` for readability. took me a minute to figure out this part haha

**Review by Yemeen — CHANGES_REQUESTED** [review]
after these looks good to me!

File: ect/ect/embed_graph.py (hunk 1 above)
Comment (lizliz) [reply ect/ect/embed_graph.py:64]: I separated this one off into another issue. I'm sure there's a more intelligent way to do this than what I was doing but I'm not ready to poke that bear today

File: ect/ect/embed_graph.py [hunk 2]
```diff
@@ -24,18 +25,25 @@ def __init__(self):
...
         Parameters:
-            vertex (str):
-                The vertex to be added.
-            x (float):
-                The x-coordinate of the vertex.
-            y (float):
-                The y-coordinate of the vertex.
-
+            vertex (hashable like int or str, or None) : The name of the vertex to add.
+            f_vertex (float) : The function value of the vertex being added.
```
[the remark this answers is not in the document]
Comment (lizliz) [reply ect/ect/embed_graph.py:33]: Ah that's supposed to be the x,y inputs. Fixed it.

File: ect/ect/embed_graph.py [hunk 3]
```diff
@@ -53,6 +61,50 @@ def add_nodes_from(self, nodes, coordinates):
...
+                    return [s[:-1] + chr(ord('A')+1+i) for i in range(num_verts)]
+                else:
+                    return (len(s)+1) * 'A'
+            else:
+                if num_verts > 1:
+                    return [s[:-1] + chr(ord(s[-1])+1+i) for i in range(num_verts)]
+                else:
+                    return len(s) * chr(ord(s[-1])+1+1)
+        else:
+            ValueError('Input must be a string or an integer')
```
[the remark this answers is not in the document]
Comment (lizliz) [reply ect/ect/embed_graph.py:106]: Fixed!

File: ect/ect/embed_graph.py [hunk 4]
```diff
@@ -146,6 +219,31 @@ def get_bounding_radius(self):
         distances = np.linalg.norm(coords - center, axis=1)
         return np.max(distances)
 
+    # ------
+    # Methods for normalizing the coordinates in various ways
+    # ------
+
+    def get_min_max_centered_coordinates(self):
```
[the remark this answers is not in the document]
Comment (lizliz) [reply ect/ect/embed_graph.py:226]: Yup that seems very reasonable, on it.

**Pull request merged**